File: catkin_ws/src/learning_service/src/turtle_command_client.cpp
//
// Created by jyh on 2020/2/23.
//
/**
 * This routine  requests the /spawn service, the service data type turtlesim:: spawn
 */

#include <ros/ros.h>
#include <std_srvs/Trigger.h>

int main(int argc, char** argv)
{
    // Initialize the ROS node
    ros::init(argc, argv, "turtle_spawn");

    // Create a node handle
    ros::NodeHandle node;


    //Once the /spawn service is found,
    // create a service client and connect to the service named /spawn
    ros::service::waitForService("/turtle_command");
    ros::ServiceClient add_turtle = node.serviceClient<std_srvs::Trigger>("/turtle_command");

    // Initialize the service data for std_srvs::Trigger
    std_srvs::Trigger srv;

    // Request service
    add_turtle.call(srv);

    // Displays the result of the response
    ROS_INFO("%s \n %s", srv.response.success?"True":"false", srv.response.message.c_str());

    return 0;
};

File: catkin_ws/src/learning_service/scripts/turtle_command_client.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# This routine  requests the /spawn service, the service data type turtlesim:: spawn

import sys
import rospy

from std_srvs.srv import Trigger, TriggerResponse

if __name__ == "__main__":

    # Initialize the ROS node
    rospy.init_node('turtle_command_client')

    # Once the /turtle_command service is found,
    rospy.wait_for_service('/turtle_command')
    try:
        # create a service client and connect to the service named /spawn
        trigger_turtle = rospy.ServiceProxy('/turtle_command', Trigger)

        # Request service , input request data
        response = trigger_turtle()
        print response.success
        print response.message
    except rospy.ServiceException, e:
        print "Service call failed: %s"%e

